tensorboard/webapp/widgets/line_chart_v2/lib/renderer/threejs_renderer.ts

```diff
@@ -92,14 +94,112 @@ function updateGeometryWithVec2Array(
   geometry.computeBoundingSphere();
 }
 
+/**
+ * Updates BufferGeometry with Float32Array that denotes flattened array of Vec2
+ * (<x, y>) representing points that form a connected set of line segments.
+ * Unlike the simpler `updatePolylineGeometry`, this variant constructs more
+ * vertices to support line thickness.
+ *
+ * This custom logic handles line thickness, since the 'linewidth' property of
+ * THREE.LineBasicMaterial is ignored [1]. Each line segment is a rectangle
+ * that is split into 2 triangles [A->B->C] and [C->B->D]. Each triangle has
+ * 3 components (x, y, z).
+ *
+ * Assuming a line segment is as follows:
+ *              A             C
+ *              |             |
+ * (startPoint) |-------------| (endPoint)
+ *              |             |
+ *              B             D
+ *
+ * [1] https://github.com/mrdoob/three.js/issues/14627
+ */
+function updateThickPolylineGeometry(
+  geometry: THREE.BufferGeometry,
+  flatVec2: Float32Array,
+  thickness: number
+) {
+  const numSegments = flatVec2.length / 2 - 1;
+  const numVertices = numSegments * 2 * 3;
+  const numComponents = numVertices * 3;
+  let positionAttributes = geometry.attributes[
+    'position'
+  ] as THREE.BufferAttribute;
+  if (!positionAttributes || positionAttributes.count !== numVertices) {
+    positionAttributes = new THREE.BufferAttribute(
+      new Float32Array(numComponents),
+      3
+    );
+    geometry.addAttribute('position', positionAttributes);
+  }
+
+  const values = positionAttributes.array as Float32Array;
+  const zeroVec = new THREE.Vector2(0, 0);
+  for (let i = 0; i < numSegments; i++) {
+    const [x1, y1, x2, y2] = [
+      flatVec2[2 * i],
+      flatVec2[2 * i + 1],
+      flatVec2[2 * i + 2],
+      flatVec2[2 * i + 3],
+    ];
+    const startPointVec = new THREE.Vector2(x1, y1);
+    const endPointVec = new THREE.Vector2(x2, y2);
+    const unitNormalVec = new THREE.Vector2(x2 - x1, y2 - y1)
```

Comment:
Hmm rotation feels a bit more crazier than what I had in mind. If we can calculate `dy / 2`, should it not be quite easy to calculate the coordinates ourselves? I am sure rotation is a simple matrix operation but doing rotation then scaling sounds like one affine transformation that _can_ be parallelized (but we don't here) and is more complex than simple arithmetic. WDYT?